sql/stringtype.go

```diff
@@ -205,38 +203,30 @@ func (t stringType) Compare(a interface{}, b interface{}) (int, error) {
 		return res, nil
 	}
 
-	var as string
-	var bs string
+	var as CollatedString
```

Comment:
This worries me.

It's fine to represent strings as CollatedStrings internally, but at the interface layer (where an integrator gives data to the engine or vice versa), we *must* retain the option to use simple golang strings.

For Row2 it's fine to define different semantics. But normal golang strings have to keep working for Row / Schema. We have people using this package in production and this will break all of them.